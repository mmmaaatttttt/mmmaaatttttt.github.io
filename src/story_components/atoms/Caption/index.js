import styled from "styled-components";
import PropTypes from "prop-types";
import { rhythm } from "utils/typography";

const Caption = styled.p`
  text-align: center;
  font-weight: 700;
  font-size: 85%;
  margin-top: ${props => props.captionMarginTop};
`;

Caption.propTypes = {
  captionMarginTop: PropTypes.string.isRequired,
};

Caption.defaultProps = {
  captionMarginTop: rhythm(0.15)
};

export default Caption;
