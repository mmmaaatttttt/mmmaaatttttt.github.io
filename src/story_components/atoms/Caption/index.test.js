import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Caption from ".";

it("renders successfully", () => {
  shallow(<Caption />);
});

it("matches snapshot", () => {
  const wrapper = shallow(
    <Caption captionMarginTop="10px">captiony mccaption</Caption>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
