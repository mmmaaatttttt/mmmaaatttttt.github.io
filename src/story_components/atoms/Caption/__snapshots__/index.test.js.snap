// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<StyledComponent
  captionMarginTop="0.27rem"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  text-align: center;
  font-weight: 700;
  font-size: 85%;
  margin-top: ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "captionMarginTop": "0.27rem",
      },
      "displayName": "styled.p",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "captionMarginTop": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "p",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`renders text when passed in successfully 1`] = `
<StyledComponent
  captionMarginTop="0.27rem"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  text-align: center;
  font-weight: 700;
  font-size: 85%;
  margin-top: ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "captionMarginTop": "0.27rem",
      },
      "displayName": "styled.p",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "captionMarginTop": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "p",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Here's a caption lolz
</StyledComponent>
`;
